.github/workflows/deploy-new-version.yml

```diff
@@ -0,0 +1,116 @@
+        uses: actions/checkout@v4
+
+      - name: Install Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: '16.14.2'
+
+      - name: Install Dependencies
+        run: npm ci
+
+      - name: Run ESLint
+        run: npm run lint
+
+      - name: Run Prettier
+        run: npm run prettier
+
+  check-version:
+    runs-on: ubuntu-latest
+    needs: format_and_lint
+
+    steps:
+      - name: Should continue
+        run: |
+          if [[ ${{ github.ref_name != 'main' && inputs.version == '' }} ]]; then
+            echo "SKIP_JOB=true" >> $GITHUB_ENV
+          fi
+
+      - name: Checkout code
+        if: ${{ env.SKIP_JOB != 'true' }}
+        uses: actions/checkout@v4
+
+      - name: Git config
+        if: ${{ env.SKIP_JOB != 'true' }}
+        run: |
+          git config user.name "${GITHUB_ACTOR}"
+          git config user.email "${GITHUB_ACTOR}@users.noreply.github.com"
+
+      - name: Install semver
+        if: ${{ env.SKIP_JOB != 'true' }}
+        run: npm install semver
+
+      - name: Get current version
+        if: ${{ env.SKIP_JOB != 'true' }}
+        run: echo "current_version=$(jq -r '.version' package.json)" >> $GITHUB_ENV
+
+      - name: Validate and set new version
+        if: ${{ env.SKIP_JOB != 'true' }}
+        run: |
+          new_version="${{ inputs.version }}"
+          current_version="${{ env.current_version }}"
+
+          if npx semver $new_version -r "<=$current_version"; then
+            echo "Error: New version ($new_version) is lowest or the same as current ($current_version)"
+            exit 1
+          fi
+
+      - name: Bump version
+        if: ${{ env.SKIP_JOB != 'true' }}
+        run: |
+          npm version ${{ inputs.version }} --no-git-tag-version
+
+      - name: Commit version change
+        if: ${{ env.SKIP_JOB != 'true' }}
+        run: |
+          git commit -am "Update version to ${{ inputs.version }}"
+          git push origin main
+
+  prepare-matrix:
+    runs-on: ubuntu-latest
+    needs: [format_and_lint, check-version]
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Determine matrix
+        id: set-matrix
+        run: |
+          if [[ "${{ github.ref_name }}" == "main" ]]; then
+            echo "matrix={\"include\":[{\"build-type\":\"production\",\"environment\":\"production-fidl\"}, {\"build-type\":\"staging\",\"tag_suffix\":\"stg\",\"environment\":\"staging-fidl\"}]}" >> $GITHUB_OUTPUT
+          else
+            echo "matrix={\"include\":[{\"build-type\":\"staging\",\"environment\":\"staging-fidl\"}]}" >> $GITHUB_OUTPUT  
+          fi
+
+  ReuseableMatrixJobForDeployment:
+    needs: 'prepare-matrix'
+    strategy:
+      matrix: ${{ fromJSON(needs.prepare-matrix.outputs.matrix) }}
+
+    uses: ./.github/workflows/release-new-version.yml
+    with:
+      version: ${{ inputs.version }}
+      environment: ${{ matrix.environment }}
+    secrets: inherit
```

Comment:
Separate note: is `secrets: inherit` here working as expected? My understanding is that it will actually pass secrets of current workflow. But current workflow is not running under any environment, so it will not actually pass anything useful. Do we actually need this?